flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/nodes/CygnusApplication.java

```diff
@@ -363,4 +368,42 @@ private void stopSinks() {
                 
     } // AgentShutdownHook
     
+    /**
+     * Yet Another Flume Supervisor. FIXME: this is a shortcut avoiding to extend the original LifecycleSupervisor
+     * class from Apache Flume, which can be be hard to do. Nevertheless, a tech debt issue has been created regarding
+     * this: https://github.com/telefonicaid/fiware-connectors/issues/354
+     */
+    private static class YAFS extends Thread {
+        
+        private final Thread[] threadArray;
+        
+        /**
+         * Constructor.
+         */
+        public YAFS() {
+            Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
+            threadArray = threadSet.toArray(new Thread[threadSet.size()]);
+        } // YAFS
+        
+        @Override
+        public void run() {
+            while (true) {
+                for (Thread t: threadArray) {
+                    // exit Cygnus if some thread (except for the main one) is found to be not alive or in a terminated
+                    // state
+                    if (!t.getName().equals("main") && (t.getState() == State.TERMINATED || !t.isAlive())) {
+                        System.exit(-1);
+                    } // if
+                } // for
+                
+                try {
+                    Thread.sleep(1000);
```

Comment:
Maybe we could extract that constant value and include it as a private final static field at the beginning of the class for easier tuning, since there are already other such constants in the class ;)